src/test/run-pass/destructure-array-2.rs

```diff
@@ -0,0 +1,75 @@
+// Copyright 2015 The Rust Project Developers. See the COPYRIGHT
```

Comment:
Could you add this test to `src/test/run-pass/dynamic-drop.rs` instead? That has a framework that tests leak-proofness in panics.